### PR TITLE
Issue-179 Implement last-modified API

### PR DESCRIPTION
**Pull request by adam-tylr**
Note the API [documentation](https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-rest.html#idp231) reads as if the "at" query parameter is optional. It says it will use the default branch. However, when writing integration tests, I always got an `com.atlassian.bitbucket.validation.ArgumentValidationException` when I excluded it saying "Streaming modifications requires a starting commit." so I made it required.

### CODE REVIEW

File: src/main/java/com/cdancy/bitbucket/rest/domain/file/LastModifiedSummary.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    LastModifiedSummary() {
+    }
+
+    @SerializedNames({ "files", "latestCommit", "errors" })
+    public static LastModifiedSummary create(final Map<String, Commit> files,
+                 final Commit latestCommit,
+                 final List<Error> errors) {
+        return new AutoValue_LastModifiedSummary(BitbucketUtils.nullToEmpty(errors),
+            files,
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/file/LastModifiedSummary.java]: Lets all use `BitbucketUtils.nullToEmpty(files)` here as well.

File: src/main/java/com/cdancy/bitbucket/rest/domain/file/LastModifiedSummary.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.cdancy.bitbucket.rest.domain.common.ErrorsHolder;
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+import java.util.Map;
+
+@AutoValue
+public abstract class LastModifiedSummary implements ErrorsHolder {
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/file/LastModifiedSummary.java]: Lets change the name to just `LastModified` to match the endpoint being used. The name here isn't horrible, it's actually fine, I just like to try and match the returning object with the endpoint name _if_ possible.

**Comment by cdancy** [comment]
@adam-tylr couple minor nitpicky things to address but all-in-all the code is clean and there is not much to pick at. Good job and thanks.

**Comment by adam-tylr** [comment]
@cdancy I agree with both of those. I even had the nulltoEmpty(files) then talked myself out of it.

**Comment by cdancy** [comment]
@adam-tylr everything looks good on my end and thanks for the PR! I'm about to step out for a few hours but if you're looking for a point release with this change I can knock that out when I come back. Let me know.

**Comment by cdancy** [comment]
I just merged this PR but noticed there were a few more things I wanted to tackle. I can do them when I get back or if you're up for it have it but here they are:

1. The fallback name should be changed to reflect the name changes above.
2. Change the endpoint/method-names from `listLastModified` to just `lastModified`

**Comment by adam-tylr** [comment]
I'll knock those out real quick. You don't need to create a release yet. I noticed the list-files api is only half implemented. You can't specify a path to list them at. I might tackle that this weekend if I have time.

**Comment by cdancy** [comment]
Another thought: I think i could probably write an HttpRequestFilter for this method that would allow us to combine both methods into a single version and if the path param is null we just scrub it from the request line. Thoughts?

**Comment by adam-tylr** [comment]
I think that makes sense. Especially since there are several endpoints all set up the same way.

**Pull request merged**